docs/developer/README.md

```diff
@@ -99,65 +99,72 @@ Integration tests use the `envtest` package from controller-runtime.  See this [
+* "multi-cluster" tests require two kind clusters;
+* "single-cluster" tests only need one kind cluster.
 
-**Note:** There are plans to add the ability to run the tests against other clusters. This is being tracked in [#112](https://github.com/k8ssandra/k8ssandra-operator/issues/112).
+The name of the test will generally indicate its type. You can also find a complete list in the GitHub action
+definitions (look for the `matrix` property in
+[kind_multicluster_e2e_tests.yaml](../../.github/workflows/kind_multicluster_e2e_tests.yaml) or
+[kind_e2e_tests.yaml](../../.github/workflows/kind_e2e_tests.yaml)).
 
-### Automated procedure
+### Resource Requirements
+Multi-cluster tests will be more resource intensive than other tests. The Docker VM used to develop and run these tests
+on a MacBook Pro is configured with 6 CPUs and 10 GB of memory. Your mileage may vary on other operating systems/setups.
 
-The makefile has a target which will create the kind clusters (deleting them first if they already exist), build the docker image and load it into both clusters before running the e2e tests.
-Just run the following:
+### Multi-cluster tests
 
-```
-make kind-e2e-test
-```
+The makefile has targets to create and configure the kind clusters (note that this automates the steps detailed at the
+beginning of this document):
 
-If you want to run a single test, set the `E2E_TEST` variable as follows:
+```shell
+# - Create kind clusters (deleting any existing ones first)
+# - Install Cert Manager and Traefik (used for ingress in e2e tests)
+make multi-create
 
+# - Build the project
+# - Generate the CRDs, then load them into the Kind cluster
+# - Build the operator image, then load it into the Kind cluster
+make multi-prepare
 ```
-make E2E_TEST=TestOperator/SingleDatacenterCluster kind-e2e-test
-```
-
-### Manual procedure 
 
-#### Create the kind clusters
-The multi-cluster tests require two clusters.
+You can then run the entire test suite:
 
+```shell
+make e2e-test
 ```
-./scripts/setup-kind-multicluster.sh --clusters 2
-```
-
-#### Build operator image
 
-Before running tests, build the operator image:
+Or a single test: 
 
-```
-make docker-build
+```shell
+make E2E_TEST=TestOperator/CreateMultiDatacenterCluster e2e-test
 ```
 
-#### Load the operator image into the clusters
-Load the operator image with `make kind-load-image`:
+Lastly, there is a target that does all of the above in a single step (this will destroy a recreate the clusters every
+time, so probably not the one to use if you're running a test repeatedly):
 
-```
-make KIND_CLUSTER=k8ssandra-0 kind-load-image
+```shell
+make E2E_TEST=TestOperator/CreateMultiDatacenterCluster kind-multi-e2e-test
 ```
 
-```
-make KIND_CLUSTER=k8ssandra-1 kind-load-image
-```
+### Single-cluster tests
 
-#### Run the tests
+Same principle, but the targets have different names:
 
-`make e2e-test` runs the tests under `test/e2e`.
+```shell
+make single-create
+make single-prepare
```

Comment:
You should replace these by `make single-up`.
It's actually an anomaly that we're not doing it in the makefile and still rely on multiple targets.

Reply:
I created `single-create` and `single-prepare`. 

I actually don't think that `single-up` is very developer friendly as it destroys and re-creates the cluster each time. If, as a developer, you only need to "refresh" the cluster by re-building the operator image and re-deploying it, that's overkill. That's where you can use just `single-prepare` it's much faster.

Reply:
... and `single-reload` is a bit overkill as well: granted, it doesn't re-create the whole cluster; but it deploys the operator images in the default `k8ssandra-operator` namespace, which is unnecessary when running e2e tests.

Reply:
kk, we can discard my comment if you prefer to use those two commands. I think the overhead of `-reload` is low and `-up` is convenient to (re)start from scratch, but it's not a hill to die on 👍